src/datachain/lib/signal_schema.py

```diff
@@ -464,19 +464,63 @@ def contains_file(self) -> bool:
         return False
 
     def slice(
-        self, keys: Sequence[str], setup: Optional[dict[str, Callable]] = None
+        self,
+        params: dict[str, Union[DataType, Any]],
+        setup: Optional[dict[str, Callable]] = None,
+        is_batch: bool = False,
     ) -> "SignalSchema":
-        # Make new schema that combines current schema and setup signals
-        setup = setup or {}
-        setup_no_types = dict.fromkeys(setup.keys(), str)
-        union = SignalSchema(self.values | setup_no_types)
-        # Slice combined schema by keys
-        schema = {}
-        for k in keys:
+        """
+        Returns new schema that combines current schema and setup signals.
+        """
+        setup_params = setup.keys() if setup else []
+        schema: dict[str, DataType] = {}
+
+        for param, param_type in params.items():
+            if param in setup_params:
+                schema[param] = str
+                continue
```

Comment:
Just check if input param is case from setup values (using `str` here because it was like this before, I suppose it is hard or not possible to get output type of lambda functions).

Reply:
> o get output type of lambda functions

output or input?